fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -2832,19 +2832,12 @@ public LoanOverdueDTO applyChargeToOverdueLoanInstallment(final Long loanId, fin
             diff = 1L;
         }
         LocalDate startDate = dueDate.plusDays(penaltyWaitPeriodValue.intValue() + 1);
-        Loan loanData = this.loanAssembler.assembleFrom(loanId);
-        LocalDate endDate = DateUtils.getBusinessLocalDate();
-        if (dueDate.isBefore(loanData.getExpectedMaturityDate())) {
-            if (chargeDefinition.feeInterval() != null) {
-                endDate = scheduledDateGenerator.getRepaymentPeriodDate(PeriodFrequencyType.fromInt(feeFrequency),
-                        chargeDefinition.feeInterval(), startDate);
-            }
-        }
+
         Integer frequencyNunber = 1;
         if (feeFrequency == null) {
             scheduleDates.put(frequencyNunber++, startDate.minusDays(diff));
         } else {
-            while (!startDate.isAfter(endDate) && !startDate.isEqual(endDate)) {
+            while (!startDate.isAfter(DateUtils.getBusinessLocalDate())) {
```

Comment:
do you want to capture for when startDate is equal to business date ?

Reply:
no it was default logic for oxygen one of implementation earlier that logic applied, now having issue in one of feature so going with default one.